test/decl/func/typed_throws.swift

```diff
@@ -216,3 +216,13 @@ struct NotAnError<T> {}
 
 func badThrowingFunctionType<T>(_: () throws(NotAnError<T>) -> ()) {}
 // expected-error@-1 {{thrown type 'NotAnError<T>' does not conform to the 'Error' protocol}}
+
+enum ObjCError: Int, Error {
+  case Others
+}
+
+import Foundation
+@objc class ObjCClass: NSObject {
+  @objc func objcTypedThrow() throws(ObjCError) -> () {}
```

Comment:
Some of Swift's platforms don't support `@objc`, so if you're going to test something with `@objc`, you'll need to do it in a test file that only runs if ObjC interop is supported.

For this change, I'll recommend adding your tests to test/attr/attr_objc.swift. There are already some tests of ordinary throws (search for `ClassThrows1`), so you should be able to add some typed throws tests somewhere around there.

(As a nice bonus, this file already has infrastructure in place to check that you've used `softenIfAccessNote()` correctly.)